geopandas/array.py

```diff
@@ -1044,25 +1044,30 @@ def fillna(self, value=None, method=None, limit=None):
             Alternatively, an GeometryArray 'value' can be given. It's expected
             that the GeometryArray has the same length as 'self'.
 
-        method : {'backfill', 'bfill', 'pad', 'ffill', None}, default None
-            Method to use for filling holes in reindexed Series
-            pad / ffill: propagate last valid observation forward to next valid
-            backfill / bfill: use NEXT valid observation to fill gap
+        method : {'backfill', 'bfill', 'ffill', None}, default None
+            Method to use for filling holes in reindexed GeoSeries:
+
+            - ffill: propagate last valid observation forward to next valid.
+            - backfill / bfill: use next valid observation to fill gap.
 
         limit : int, default None
             If method is specified, this is the maximum number of consecutive
             NaN values to forward/backward fill. In other words, if there is
             a gap with more than this number of consecutive NaNs, it will only
             be partially filled. If method is not specified, this is the
             maximum number of entries along the entire axis where NaNs will be
-            filled.
+            filled. Must be greater than 0 if not None.
 
         Returns
         -------
         GeometryArray
         """
-        if method is not None:
-            raise NotImplementedError("fillna with a method is not yet supported")
+        from pandas.core.missing import clean_fill_method
+
+        if value is None:
+            return super().fillna(method=clean_fill_method(method), limit=limit)
```

Comment:
Is the usage of `clean_fill_method` needed here? Ideally we don't have to use anything from `pandas.core` since that's not public API

Reply:
`return super().fillna(method=method, limit=limit)` can handle this.
If the method isn't right, there will get an error.